aten/src/ATen/native/SpectralOps.cpp

```diff
@@ -157,7 +174,12 @@ Tensor fft_c2c(Tensor input, c10::optional<int64_t> n_opt,
     input = resize_fft_input(input, dim, n);
   }
   const auto norm = norm_from_string(norm_str, forward);
-  return at::_fft_c2c(input, dim, static_cast<int64_t>(norm), forward);
+  if (out.defined()) {
+    TORCH_CHECK(out.is_complex(), "fft expects a complex output tensor, but got ", out.scalar_type());
```

Comment:
"fft" here is interesting because this warning could be triggered by many operations. 

In the above warning a capital "FFT" is used, which doesn't seem much better.

Maybe this function should accept a string with the appropriate function name to display? So it could say, for example, "fft2 expects..." when `torch.fft.fft2` is called?

I guess c2r and r2c already have enough logic to disambiguate which function called them, but they could adopt this approach for consistency, too.

Reply:
I've added the name argument but  `fft2` will still say `fftn` in error messages because I'm calling `native::fftn` rather than `fftn_c2c` which expects slightly different arguments.

Reply:
I think we can live with that.